ext/net/quic.rs

```diff
@@ -658,3 +684,336 @@ pub(crate) fn op_quic_set_send_stream_priority(
     None => Err(generic_error("Unable to set priority")),
   }
 }
+
+#[op2(fast)]
+#[bigint]
+pub(crate) fn op_quic_get_send_stream_id(
+  state: Rc<RefCell<OpState>>,
+  #[smi] rid: ResourceId,
+) -> Result<u64, AnyError> {
+  let resource = state
+    .borrow()
+    .resource_table
+    .get::<SendStreamResource>(rid)?;
+  let r = RcRef::map(resource, |r| &r.0).try_borrow();
+  match r {
+    Some(s) => Ok(quinn::VarInt::from(s.id()).into_inner()),
+    None => Err(generic_error("Unable to get id")),
+  }
+}
+
+#[op2(fast)]
+#[bigint]
+pub(crate) fn op_quic_get_recv_stream_id(
+  state: Rc<RefCell<OpState>>,
+  #[smi] rid: ResourceId,
+) -> Result<u64, AnyError> {
+  let resource = state
+    .borrow()
+    .resource_table
+    .get::<RecvStreamResource>(rid)?;
+  let r = RcRef::map(resource, |r| &r.0).try_borrow();
+  match r {
+    Some(s) => Ok(quinn::VarInt::from(s.id()).into_inner()),
+    None => Err(generic_error("Unable to get id")),
+  }
+}
+
+pub(crate) mod webtransport {
+  // MIT License
+  //
+  // Copyright (c) 2023 Luke Curley
+  //
+  // Permission is hereby granted, free of charge, to any person obtaining a copy
+  // of this software and associated documentation files (the "Software"), to deal
+  // in the Software without restriction, including without limitation the rights
```

Comment:
Is there a reason to vendor this code? If so can you please add a link here to the original code?